ts-obs/tests/metrics_test.go

```diff
@@ -0,0 +1,212 @@
+package tests
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+func testRetentionSetDefault(t testing.TB, period int) {
+	var set *exec.Cmd
+
+	t.Logf("Running 'ts-obs metrics retention set-default %d'\n", period)
+	set = exec.Command("ts-obs", "metrics", "retention", "set-default", strconv.Itoa(period))
+
+	out, err := set.CombinedOutput()
+	if err != nil {
+		t.Logf(string(out))
+		t.Fatal(err)
+	}
+}
+
+func testRetentionSet(t testing.TB, metric string, period int) {
+	var set *exec.Cmd
+
+	t.Logf("Running 'ts-obs metrics retention set %v %d'\n", metric, period)
+	set = exec.Command("ts-obs", "metrics", "retention", "set", metric, strconv.Itoa(period))
+
+	out, err := set.CombinedOutput()
+	if err != nil {
+		t.Logf(string(out))
+		t.Fatal(err)
+	}
+}
+
+func testRetentionReset(t testing.TB, metric string) {
+	var reset *exec.Cmd
+
+	t.Logf("Running 'ts-obs metrics retention reset %v'\n", metric)
+	reset = exec.Command("ts-obs", "metrics", "retention", "reset", metric)
+
+	out, err := reset.CombinedOutput()
+	if err != nil {
+		t.Logf(string(out))
+		t.Fatal(err)
+	}
+}
+
+func testChunkIntervalSetDefault(t testing.TB, interval string) {
+	var set *exec.Cmd
+
+	t.Logf("Running 'ts-obs metrics chunk-interval set-default %v'\n", interval)
+	set = exec.Command("ts-obs", "metrics", "chunk-interval", "set-default", interval)
+
+	out, err := set.CombinedOutput()
+	if err != nil {
+		t.Logf(string(out))
+		t.Fatal(err)
+	}
+}
+
+func testChunkIntervalSet(t testing.TB, metric, interval string) {
+	var set *exec.Cmd
+
+	t.Logf("Running 'ts-obs metrics chunk-interval set %v %v'\n", metric, interval)
+	set = exec.Command("ts-obs", "metrics", "chunk-interval", "set", metric, interval)
+
+	out, err := set.CombinedOutput()
+	if err != nil {
+		t.Logf(string(out))
+		t.Fatal(err)
+	}
+}
+
+func testChunkIntervalReset(t testing.TB, metric string) {
+	var reset *exec.Cmd
+
+	t.Logf("Running 'ts-obs metrics chunk-interval reset %v'\n", metric)
+	reset = exec.Command("ts-obs", "metrics", "chunk-interval", "reset", metric)
+
+	out, err := reset.CombinedOutput()
+	if err != nil {
+		t.Logf(string(out))
+		t.Fatal(err)
+	}
+}
+
+func verifyRetentionPeriod(t testing.TB, metric string, expectedDuration time.Duration) {
+	var durS int
+	var dur time.Duration
+
+	portforward := exec.Command("kubectl", "--namespace", "default", "port-forward", "ts-obs-timescaledb-0", "5433:5432")
+	err := portforward.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(10 * time.Second)
+
+	os.Setenv("PGPASSWORD_POSTGRES", "tea")
+
+	pool, err := pgxpool.Connect(context.Background(), "postgres://postgres:"+os.Getenv("PGPASSWORD_POSTGRES")+"@localhost:5433/postgres")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = pool.QueryRow(context.Background(),
+		`SELECT EXTRACT(epoch FROM _prom_catalog.get_metric_retention_period($1))`,
```

Comment:
I wonder if we should have a cli command to get the retention period for a metric